deployer/deployer-services/src/main/java/io/hyscale/deployer/services/util/K8sPodUtil.java

```diff
@@ -278,4 +283,152 @@ private static String validateAndGetInitContainerMessage(List<V1ContainerStatus>
 		}
 		return initContainerStatus;
 	}
+	
+	/**
+	 * 
+	 * @param podList
+	 * @return pod owner kind if all pods have same owner, else null
+	 */
+	
+	public static String getPodsOwner(List<V1Pod> podList) {
+	    if (podList == null || podList.isEmpty()) {
+	        return null;
+	    }
+	    
+	    String podOwner = getPodOwner(podList.get(0));
+	    if (StringUtils.isBlank(podOwner)) {
+	        return null;
+	    }
+	    
+	    for (V1Pod pod: podList) {
+	        if (!podOwner.equals(getPodOwner(pod))) {
+	            return null;
+	        }
+	    }
+	    
```

Comment:
How is this justified ?

Reply:
If all the pods have same owner kind we return it, else we return null